asb-native/src/main/java/org/ballerinax/asb/util/ASBUtils.java

```diff
@@ -42,14 +43,37 @@ public class ASBUtils {
      * @return Converted BMap object.
      */
     public static BMap<BString, Object> toBMap(Map<String, Object> map) {
-        BMap<BString, Object> returnMap = ValueCreator.createMapValue();
+        MapType mapType = TypeCreator.createMapType(PredefinedTypes.TYPE_ANY);
+        BMap<BString, Object> envMap = ValueCreator.createMapValue(mapType);
         if (map != null) {
             for (Object aKey : map.keySet().toArray()) {
-                returnMap.put(StringUtils.fromString(aKey.toString()),
-                        StringUtils.fromString(map.get(aKey).toString()));
+                Object value = map.get(aKey);
+                String classType = value.getClass().getName();
+                if (classType.equals("java.lang.String")) {
```

Comment:
Shall we check if we can improve readability by adding  `switch-case` construct instead of `if-else` ladder.